measure_optimize.py

```diff
@@ -2,180 +2,2412 @@
+                        # initialize grey-box value
+                        self.mod.my_block.egb.inputs[grey_box_name] = new_fim[a][b]
+
+        # initialize determinant
+        # use slogdet to avoid ill-conditioning issue
+        _, logdet = np.linalg.slogdet(new_fim)
+
+        # if D-optimality
+        if grey_box:
+            # initialize grey-box module output
+            self.mod.my_block.egb.outputs["log_det"] = logdet
+
+        if self.optimize_print_level >= 1:
+            print("Warmstart initialize FIM with: ", new_fim)
+            print("Warmstart logdet:", logdet)
+            print("Warmstart eigen value:", np.linalg.eigvals(new_fim))
+
+        # manual and dynamic install initial
+        def total_dynamic_exp(m):
+            """compute the number of time points selected in total, to initialize total_number_dynamic_measurements
+            This value is for initializing the total manual number constraint
+            """
+            return sum(
+                m.cov_y[i, i].value
+                for i in range(
+                    self.n_static_measurements, self.num_measure_dynamic_flatten
+                )
+            )
+
+        def total_exp(m):
+            """compute the number of all measurements selected in total, to initialize total_number_measurements
+            This value is for initializing integer cut
+            """
+            return sum(
+                m.cov_y[i, i].value for i in range(self.num_measure_dynamic_flatten)
+            )
+
+        ### cost constraints
+        def cost_compute(m):
+            """Compute cost
+            cost = static-cost measurement cost + dynamic-cost measurement installation cost + dynamic-cost meausrement timepoint cost
+            """
+            return sum(
+                m.cov_y[i, i].value * self.cost_list[i]
+                for i in range(self.num_measure_dynamic_flatten)
+            ) + sum(
+                m.if_install_dynamic[j].value
+                * self.dynamic_install_cost[j - self.n_static_measurements]
+                for j in m.dim_dynamic
+            )
+
+        # compute cost
+        cost_init = cost_compute(self.mod)
+        # copmute total number of dynamic time points selected
+        total_dynamic_initial = total_dynamic_exp(self.mod)
+        # compute total number of measurements selected
+        total_initial = total_exp(self.mod)
+
+        # initialize model variables
+        self.mod.total_number_dynamic_measurements.value = total_dynamic_initial
+        self.mod.total_number_measurements.value = total_initial
+        self.mod.cost.value = cost_init
+
+        if self.optimize_print_level >= 1:
+            print("warmstart initialize total measure:", total_initial)
+            print("warmstart initialize total dynamic: ", total_dynamic_initial)
+            print("warmstart initialize cost:", cost_init)
+
+    def continuous_optimization_cvxpy(self, objective="D", budget=100, solver=None):
```

Comment:
Please debug/improve this function as a separate branch/PR.

Reply:
created a separated branch for this.